include/vcpkg/base/files.h

```diff
@@ -186,19 +188,56 @@ namespace vcpkg
     struct ReadFilePointer : FilePointer
     {
         ReadFilePointer() = default;
+        ReadFilePointer(ReadFilePointer&&) = default;
         explicit ReadFilePointer(const Path& file_path, std::error_code& ec) noexcept;
 
+        ReadFilePointer& operator=(ReadFilePointer&& other) noexcept
+        {
+            ReadFilePointer fp{std::move(other)};
+            std::swap(m_fs, fp.m_fs);
+            return *this;
+        }
+
         size_t read(void* buffer, size_t element_size, size_t element_count) const noexcept
         {
             return ::fread(buffer, element_size, element_count, m_fs);
         }
+
+        int getc() const noexcept { return ::fgetc(m_fs); }
+    };
+
+    struct PositionedReadPointerU32
```

Comment:
To other reviewers: should we make ReadFilePointer always act like this? I would want to cut down on the number of available seeks if so.